crates/re_ui/src/lib.rs

```diff
@@ -89,6 +89,51 @@ pub fn apply_style_and_install_loaders(egui_ctx: &egui::Context) {
     );
 
     design_tokens().apply(egui_ctx);
+
+    egui_ctx.style_mut(|style| {
+        style.number_formatter = egui::style::NumberFormatter::new(format_with_decimals_in_range);
+    });
+}
+
+fn format_with_decimals_in_range(
+    value: f64,
+    decimal_range: std::ops::RangeInclusive<usize>,
+) -> String {
+    fn format_with_decimals(value: f64, decimals: usize) -> String {
+        re_format::FloatFormatOptions::DEFAULT_f64
+            .with_decimals(decimals)
+            .with_strip_trailing_zeros(false)
+            .format(value)
+    }
+
+    let epsilon = 16.0 * f32::EPSILON; // margin large enough to handle most peoples round-tripping needs
+
+    let min_decimals = *decimal_range.start();
+    let max_decimals = *decimal_range.end();
+    debug_assert!(min_decimals <= max_decimals);
+    debug_assert!(max_decimals < 100);
+    let max_decimals = max_decimals.min(16);
+    let min_decimals = min_decimals.min(max_decimals);
```

Comment:
nit: the presence of `at_most`/`at_least` has made me weak :)
```suggestion
    let max_decimals = max_decimals.at_most(16);
    let min_decimals = min_decimals.at_most(max_decimals);
```